contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -3328,25 +3336,47 @@ SELECT
 WHERE FALSE;
 GRANT SELECT ON sys.credentials TO PUBLIC;
 
-CREATE VIEW sys.sql_logins AS
+CREATE OR REPLACE VIEW sys.sql_logins AS 
 SELECT
-    CAST(NULL as sys.sysname) AS name,
-    CAST(NULL as INT) AS principal_id,
-    CAST(NULL as sys.VARBINARY(85)) AS sid,
-    CAST(NULL as sys.BPCHAR(1)) AS type,
-    CAST(NULL as sys.nvarchar(60)) AS type_desc,
-    CAST(NULL as INT) AS is_disabled,
-    CAST(NULL as sys.DATETIME) AS create_date,
-    CAST(NULL as sys.DATETIME) AS modify_date,
-    CAST(NULL as sys.sysname) AS default_database_name,
-    CAST(NULL as sys.sysname) AS default_language_name,
-    CAST(NULL as INT) AS credential_id,
-    CAST(NULL as INT) AS owning_principal_id,
-    CAST(0 as sys.BIT) AS is_fixed_role,
-    CAST(0 as sys.BIT) AS is_policy_checked,
-    CAST(0 as sys.BIT) AS is_expiration_checked,
-    CAST(NULL as sys.varbinary(256)) AS password_hash
-WHERE FALSE;
+  CAST(Ext.orig_loginname AS sys.SYSNAME) AS name,
+  CAST(Base.oid As INT) AS principal_id,
+  CAST(CAST(Base.oid as INT) as sys.varbinary(85)) AS sid,
+  CAST(Ext.type AS sys.BPCHAR(1)) COLLATE sys.database_default AS type, 
+  CAST(
+    CASE
+      WHEN Ext.type = 'S' THEN 'SQL_LOGIN' 
+      WHEN Ext.type = 'R' THEN 'SERVER_ROLE' 
+      WHEN Ext.type = 'U' THEN 'WINDOWS_LOGIN' 
+      ELSE NULL 
+    END 
```

Comment:
We have a condition below `AND Ext.type = 'S'`, so these cases are invalid. `type_desc` can only be S